### ISSUE TITLE
Create method `Faker.Random.Numeric`

### ISSUE THREAD

**Issue by mniak** [bchavez/Bogus#223]
It would create a number 0-padded in the left. Shortcut for
```
faker.Random.Number(Math.Pow(10, x)).ToString(new string('0', x))
```

**Comment by bchavez** [comment]
Hi Andre,

Thank you for your feedback. I appreciate it.

For now, I think you'll probably want to create a custom extension method on Bogus' API to handle your pad-left scenario. Here's how you could do it:

```csharp
void Main()
{
   var faker = new Faker();
   
   faker.Random.PaddedNumber(999999, 6).Dump();
}

public static class MyCustomBogusExtensions
{
   public static string PaddedNumber(this Randomizer r, int maxNumber, int sLength){
      var digits = Math.Log10(maxNumber);
      
      if( digits >= sLength )
         throw new ArgumentOutOfRangeException("number of digits must be less than or equal the result string length.");

      var value = r.Number(maxNumber);
      
      return value.ToString().PadLeft(sLength, '0');
   }
}
```

Alternatively, Bogus has a built-in extension called `.ClampLength` that kind-of does the same thing, but except `.ClampLength` pads-right:

```csharp
var faker = new Faker();
faker.Random.Number(1234).ToString().ClampLength(5, 5, '0')
```

I think you'll have to rely on a custom C# extension method in your app as demonstrated above, but I will keep it in mind if others request the same feature; then we can look to add it to Bogus.

Thanks,
Brian

:evergreen_tree: :crystal_ball: **[PINES - Fate](https://www.youtube.com/watch?v=M2D7lCeVxGs)**

**Comment by mniak** [comment]
I only created the issue because I intend to contribute. 

What do you think about the suggestion? 
Am I welcome to contribute? 
Should it be named `Numeric` or `StringDigits` or anything else?

**Comment by bchavez** [comment]
Hi Andre,

I think your suggestion is good. I can see why you would need it. The extension could complement the [`.ClampLength` extension](https://github.com/bchavez/Bogus/blob/master/Source/Bogus/Extensions/ExtensionsForString.cs) that we already have in Bogus.

In retrospect, maybe I should have named `.ClampLength()`, `.ClampStringRight()` and paired it with `.ClampStringLeft()`. Or maybe `.PadLeftFixedLength()` or `.PadRightFixedLength()`. I don't know.

However, with new API additions and API changes, I try to wait until I have more than 1 person ask for this feature. When possible, I try to avoid adding extension methods like this that are not within the immediate scope of Bogus.

A few helper methods to make using Bogus easier is okay, but I don't want to create too many extension methods for types I don't own.

The ***problem*** I see with extension methods like this is the ambiguity of behavior. What should we do when the number of digits exceeds the length of the fixed string? IE: Number `1234567` with a fixed string length of 4? Should we throw an exception because the number is too big to fit in a string of 4 characters? or do we allow the code to continue without error and just give back a string with length 7?

I don't know. It depends on your requirements.

This is why I think the workaround is acceptable since all you need is a basic C# extension method class in your project that fits your exact need.

-----

Of course, you are welcome to contribute (and I very much appreciate your enthusiasm), but I feel this particular feature is better left to be implemented as an extension method in your project for the reasons stated above. It's a specific case of needing an integer padded left with zeros in a fixed length string. If more people ask for this feature or find it helpful, then maybe will consider making changes.

But for now, I would like to wait.

Hope that helps.

Thanks,
Brian

:walking_man: :walking_woman: **[Missing Persons - Walking in L.A.](https://www.youtube.com/watch?v=R_UpLtGEWoY)**
